reactions/src/SPT.cpp

```diff
@@ -652,8 +651,29 @@ static double ploopn2_mgdd( const PowerSpectrum& P_L, double vars[], double k, d
     return pow2(r)*2.*P_L(k*r)*(P_L(kargs[0])*p22 + 3.*P_L(k)*p13);
 }
 
+// BILL MOD
+static double ploopn2_mgdd_nu( const PowerSpectrum& P_L, double vars[], double k, double r, double x){
+  double kargs[4],kv[3],xv[3], p22,p13;
+  IOW iow;
+        kv[0] = k;
+        kv[1] = k*r;
+        kv[2] = kv[1];
+        xv[0] = -0.99999999; // don't set to -1 otherwise solver encounters singularity
+        xv[1] = x;
+        xv[2] = -x;
+        kargs[0] = sqrt(kv[1]*kv[1]+kv[0]*kv[0]-2.*kv[1]*kv[0]*xv[1]);
```

Comment:
maybe put a comment here re what this actually is following the notes (especially useful if these expressions go into the paper)

Reply:
So a comment that kargs[0] corresponds to k*sqrt(1+r^2-2rx) in the Overleaf notation?

Reply:
something like that, yes -- but it's not necessary or urgent, I'm just commenting as I go :-) we can chat about these suggestions with Ben too (I certainly don't have a complete overview of the code, yet)

Reply:
I'll leave this one for the time being. For now have added comments in many places indicating what our integrated k-vector is whicch hopefully makes such expressions more understandable.